### PR TITLE
add reusable workflows for deploying nightly binaries

### PR DESCRIPTION
**Pull request by samansmink**
This PR adds the reusable workflow for extension deployment to the postgress scanner, which allows producing nightly binaries. In the process, I removed the workflows that would be duplicates, leaving only the linux job that would actually run the tests with the postgres server. Also i added a require-env to all tests because the reusable workflow will call `make test`

@Mytherin Before merging we need to add the s3 credentials and signing token to the repo. Maybe we can add them as an [organisation secret](https://docs.github.com/en/codespaces/managing-codespaces-for-your-organization/managing-secrets-for-your-repository-and-organization-for-github-codespaces#adding-secrets-for-an-organization)?

### CODE REVIEW

File: .github/workflows/MainDistributionPipeline.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#
+# This workflow calls the main distribution pipeline from DuckDB to build, test and (optionally) release the extension
+#
+name: Main Extension Distribution Pipeline
+on:
+  push:
+  pull_request:
```
Comment (Mytherin) [inline .github/workflows/MainDistributionPipeline.yml:7]: Should this run on pull requests as well?

**Review by Mytherin — COMMENTED** [review]
Thanks for the PR! Looks great. One minor comment. For the organization secrets, perhaps @hannes can have a look?

**Comment by samansmink** [comment]
> Thanks for the PR! Looks great. One minor comment. For the organization secrets, perhaps @hannes can have a look?

I'll chat with hannes today on this

File: .github/workflows/MainDistributionPipeline.yml (hunk 1 above)
Comment (samansmink) [reply .github/workflows/MainDistributionPipeline.yml:7]: I think so? I removed most of the other jobs so without it, we would only test on linux gcc4 for PRs. I could add a `needs` to make it depend on the `linux-tests-postgres` to reduce ci time for broken builds?

**Comment by Mytherin** [comment]
Thanks for the updates - is this ready to merge?

**Comment by samansmink** [comment]
yea think so!

**Pull request merged**